File: 2015/Day2.rb
# Part 1
total = 0
File.readlines('Day2_input.txt').each do |b|
    sides = []
    sides << b.split('x')[0].to_i
    sides << b.split('x')[1].to_i
    sides << b.split('x')[2].to_i
    sides = sides.sort
    # puts sides.sort, '-----'
    area = (3*sides[0]*sides[1]) + (2*sides[1]*sides[2]) + (2*sides[0]*sides[2]) #+ (sides[0]*sides[1])
    # puts "#{b} area = " + area.to_s
    total += area
end
puts "Part 1 = #{total}"


# Part 2
total = 0
File.readlines('Day2_input.txt').each do |b|
    sides = []
    sides << b.split('x')[0].to_i
    sides << b.split('x')[1].to_i
    sides << b.split('x')[2].to_i
    sides = sides.sort
    ribbon = (2*sides[0]) + (2*sides[1]) + (sides[0]*sides[1]*sides[2])
    total += ribbon
end
puts "Part 2 = #{total}"

File: 2015/Day2.py
import numpy as np

# Part 1
total_area = 0
area = 0

for box in open("day2_input.txt").readlines():
  # print ("Box = ", box)
  a,b ,c = np.sort([int(dim) for dim in box.split('x')])
  area = (3 * a * b) + (2 * b * c) + (2 * a * c)
  total_area += area

print("Part 1 =", total_area)

# Part 2
ribbon = 0
bow = 0

for box in open("day2_input.txt").readlines():
  a, b, c = np.sort([int(dim) for dim in box.split('x')])
  ribbon += (2 * a) + (2 * b)
  bow += a * b * c

print("Part 2 = ribbon + bow = ", ribbon, "+", bow, " = ", ribbon+bow)
